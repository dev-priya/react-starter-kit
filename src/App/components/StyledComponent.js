import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// styled-component example
const StyledDiv = styled.div`
  border: 1px solid gray;
  border-radius: 4px;
  padding: 4px;
  margin-top: 8px;
  display: inline-block;
  cursor: pointer;

  &:hover {
    background-color: #f1c40f;
  }
`;

// you need this for finding this item during unit testing
StyledDiv.displayName = 'StyledDiv';

const StyledComponent = ({ message }) => <StyledDiv>{message}</StyledDiv>;

StyledComponent.propTypes = {
  message: PropTypes.string.isRequired
};

export default StyledComponent;