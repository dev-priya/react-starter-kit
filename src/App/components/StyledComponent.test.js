import React from 'react';
import {shallow} from 'enzyme';
import StyledComponent from './StyledComponent';

describe('StyledComponent', () => {
  const wrapper = shallow(<StyledComponent message="test message" />);

  it('renders a StyledDiv', () => {
    expect(wrapper.find('StyledDiv').length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});