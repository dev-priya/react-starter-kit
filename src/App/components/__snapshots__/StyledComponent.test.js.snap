// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledComponent matches the snapshot 1`] = `
<StyledDiv>
  test message
</StyledDiv>
`;
